oioioi/contests/utils.py

```diff
@@ -247,6 +256,96 @@ def visible_rounds(request):
     return [r for r in queryset if controller.can_see_round(request, r)]
 
 
+@request_cached
+def get_number_of_rounds(request):
+    """Returns the number of rounds in the current contest.
+    """ 
+    return Round.objects.filter(contest=request.contest).count()
+
+
+def get_contest_dates(request):
+    """Returns the end_date of the latest round and the start_date
+    of the earliest round.
+    """ 
+    if isinstance(request, HttpRequest):
+        rtimes = rounds_times(request, request.contest)
+    else:
+        rtimes = generic_rounds_times(None, request.contest)
+
+    ends = [
+        rt.get_end()
+        for rt in rtimes.values()
+    ]
+    starts = [
+        rt.get_start()
+        for rt in rtimes.values()
+    ]
+
+    if (not starts or starts.count(None)):
+        min_start = -1
+    else:
+        min_start = min(starts)
+
+    if (not ends or ends.count(None)):
+        max_end = -1
+    else:
+        max_end = max(ends)
+
+    return min_start, max_end
+
+
+@request_cached
+def get_problems_sumbmission_limit(request):
+    """Returns the list of distinct submission limits in the current contest.
+    """
+    controller = request.contest.controller
+    queryset = (
+        ProblemInstance.objects.filter(contest=request.contest)
+        .prefetch_related('round')
+    )
+
+    if (queryset is None):
+        return None
+    
+    limits = set()
+    for p in queryset:
+        limits.add(controller.get_submissions_limit(request, p))
+
+    return sorted(limits)
+
+
+@request_cached
+def get_results_visibility(request):
+    """Returns the results visibility for each round in the contest"""
+    if isinstance(request, HttpRequest):
+        rtimes = rounds_times(request, request.contest)
+    else:
+        rtimes = generic_rounds_times(None, request.contest)
+
+    dates = list()
+    for r in rtimes.keys():
+        results_date = rtimes[r].results_date()
+        public_results_date = rtimes[r].public_results_date()
+ 
+        if (results_date is None or results_date <= request.timestamp):
+            results_visibility = 'immediately'
+        else:
+            results_visibility = f'after {results_date.strftime("%Y-%m-%d %H:%M:%S")}'
+
+        if (public_results_date is None or public_results_date <= request.timestamp):
+            public_results_visibility = 'immediately'
+        else:
+            public_results_visibility = f'after {public_results_date.strftime("%Y-%m-%d %H:%M:%S")}'
```

Comment:
These strings should be translated.

Reply:
As far as I understand, in the recent commit I fixed it so that they're translated in a template - but in case I'm wrong, please let me know!

Reply:
Unfortunately, `{{ blocktrans with value as var }}` doesn't translate `value`, it only conditionalizes the string to translate based on the contents of `value`, which in this basic case means just inserting the `value` string in appropriate places.
I myself wasn't sure if that was the case, but running `cd /sio2/oioioi && /sio2/deployment/manage.py makemessages -a` in the web container shows no trace of these strings in `.po` files.